### PR TITLE
tools/misc: replace name with pname&version

### PR DESCRIPTION
**Pull request by Stunkymonkey**
###### Motivation for this change
#103997

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by davidak** [comment]
this will trigger a mass rebuild and should get merged to staging

**Comment by Stunkymonkey** [comment]
i prefer to identify the packages causing the massive rebuilds and separate them to get merged in `staging`. The rest still merge into `master`

**Comment by davidak** [comment]
Result of `nixpkgs-review pr 132359` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)

**Comment by davidak** [comment]
I did not know how to find out which package causes the most rebuilds, so i did some research and put together this:

```
# check out pull request
gh pr checkout 132359

# get number of changed files from github api, up to 100
[davidak@gaming:~/code/nixpkgs]$ for file in $(curl -H "Accept: application/vnd.github.v3+json" https://api.github.com/repos/NixOS/nixpkgs/pulls/132359/files?per_page=100 | jq -r '.[] | (.filename)'); do echo $file; done | wc -l
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 56576    0 56576    0     0   131k      0 --:--:-- --:--:-- --:--:--  131k
54

# get pname from changed files
[davidak@gaming:~/code/nixpkgs]$ for file in $(curl -H "Accept: application/vnd.github.v3+json" https://api.github.com/repos/NixOS/nixpkgs/pulls/132359/files?per_page=100 | jq -r '.[] | (.filename)'); do grep -Po 'pname = "\K.*(?=";)' $file; done
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 56576    0 56576    0     0  97042      0 --:--:-- --:--:-- --:--:-- 97042
apparix
aws-mturk-clt
bfr
bogofilter
ccze
convmv
dbacl
detox
dmg2img
dvtm
dvtm-unstable
ent
fasd
figlet
fileschanged
git-fire
jdiskreport
kermit
kisslicer
logtop
lrzsz
makebootfat
mysql2pgsql
odroid-xu3-bootloader
pal
pg_top
pk2cmd
proxytunnel
pv
qjoypad
renameutils
riemann-c-client
rkflashtool
rrdtool
sixpair
slsnif
smc
t1utils
timidity
tmpwatch
tmux-cssh
ttf2pt1
ttf-mkfontdir
unclutter
uucp
uudeview
vorbisgain
wv2
xdxf2slob
xflux
xtitle
yubico-piv-tool
yubikey-personalization-gui

# get number of mentions of those packages in nixpkgs
[davidak@gaming:~/code/nixpkgs]$ for file in $(curl -H "Accept: application/vnd.github.v3+json" https://api.github.com/repos/NixOS/nixpkgs/pulls/132359/files?per_page=100 | jq -r '.[] | (.filename)'); do grep -Po 'pname = "\K.*(?=";)' $file | xargs -I {} sh -c 'echo -n "{}: " >/dev/stderr; grep --text "{}" --exclude-dir=.git . -R | wc -l'; done
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 56576  100 56576    0     0   445k      0 --:--:-- --:--:-- --:--:--  445k
apparix: 4
aws-mturk-clt: 3
bfr: 326
bogofilter: 8
ccze: 3
convmv: 3
dbacl: 4
detox: 6
dmg2img: 5
dvtm: 17
dvtm-unstable: 2
ent: 509342
fasd: 19
figlet: 37
fileschanged: 11
git-fire: 5
jdiskreport: 10
kermit: 4
kisslicer: 5
logtop: 5
lrzsz: 11
makebootfat: 3
mysql2pgsql: 3
odroid-xu3-bootloader: 2
pal: 1494
pg_top: 9
pk2cmd: 5
proxytunnel: 4
pv: 10361
qjoypad: 4
renameutils: 10
riemann-c-client: 5
rkflashtool: 6
rrdtool: 30
sixpair: 9
slsnif: 4
smc: 274
t1utils: 12
timidity: 35
tmpwatch: 6
tmux-cssh: 8
ttf2pt1: 18
ttf-mkfontdir: 1
unclutter: 140
uucp: 38
uudeview: 9
vorbisgain: 13
wv2: 167
xdxf2slob: 4
xflux: 16
xtitle: 4
yubico-piv-tool: 4
yubikey-personalization-gui: 12

# now sort them
[davidak@gaming:~/code/nixpkgs]$ cat x | sort -t' ' -k2 -n -r
ent: 509342 (actually 1)
pv: 10361 (5)
pal: 1494 (1)
bfr: 326 (1)
smc: 274 (2)
wv2: 167 (1)
unclutter: 140 (6)
uucp: 38
figlet: 37
timidity: 35
rrdtool: 30
fasd: 19
ttf2pt1: 18
dvtm: 17
xflux: 16
vorbisgain: 13
yubikey-personalization-gui: 12
t1utils: 12
lrzsz: 11
fileschanged: 11
renameutils: 10
jdiskreport: 10
uudeview: 9
sixpair: 9
pg_top: 9
tmux-cssh: 8
bogofilter: 8
tmpwatch: 6
rkflashtool: 6
detox: 6
riemann-c-client: 5
pk2cmd: 5
logtop: 5
kisslicer: 5
git-fire: 5
dmg2img: 5
yubico-piv-tool: 4
xtitle: 4
xdxf2slob: 4
slsnif: 4
qjoypad: 4
proxytunnel: 4
kermit: 4
dbacl: 4
apparix: 4
mysql2pgsql: 3
makebootfat: 3
convmv: 3
ccze: 3
aws-mturk-clt: 3
odroid-xu3-bootloader: 2
dvtm-unstable: 2
ttf-mkfontdir: 1
```

It's not perfect since for example the string 'ent' is included everywhere. In 'gtkClient', 'ancient warfare' and 'Efficient learning'.

When i grep for ' ent ' i only get 2 results which is the package itself.

This method also only tells with packages these packages are a dependency of, but not of with packages those packages are dependencies.

tl;dr this doesn't answer the question

but i used this to find out which packages do get build

`nixpkgs-review pr --skip-package-regex '' 132359`

184 packages updated:
adapta-gtk-theme almanah altair-v4.0.2 apparix appimage-run appimage-run-tests Apple.Music asciidoc-full asciidoc-full asciidoc-full-with-plugins aws-mturk-clt betterdiscord-installer bfr blockbench-electron bogofilter btrbk ccache ccache-links-wrapper- ccze chain-desktop-wallet chrome-gnome-shell chrysalis clevis collectd collectd-data convmv crip dbacl deltachat-electron denemo detox devdocs-desktop disorderfs dmg2img dvtm dvtm-unstable ElectronMail electronplayer elementary-session-settings-unstable ent evolution evolution-ews evolution-with-plugins fancy-motd-unstable fasd-unstable figlet fileschanged flacon flexoptix-app freetube frescobaldi git-fire-unstable gnome-session gnome-shell gnome-shell-extension-EasyScreenCast-unstable gnome-shell-extension-gsconnect gnome-shell-extension-night-theme-switcher gnome-shell-pomodoro gnome-terminal gnome-tweaks ino irccloud jdiskreport jitsi-meet-electron joplin-desktop kcollectd keeweb kermit kisslicer lbry-desktop ledger-live-desktop lens libvmi lilypond lilypond lilypond-with-fonts linuxband logseq logtop losslesscut lrzsz luksmeta lunar-client makebootfat marktext-v0.16.2-binary mathpix-snipping-tool midisheetmusic minetime minicom mma moc molotov mrtg munin MyCrypto mysql2pgsql neo4j-desktop notable ntopng nuclear-v0.6.6 octant-desktop p3x-onenote pal Patchwork pcloud pg_top phosh picocom pk2cmd plexamp pv python3.8-labgrid python3.8-xmodem python3.9-labgrid python3.9-xmodem qemu-xen-host-cpu-only qemu-xen-host-cpu-only qemu-xen-host-cpu-only qemu-xen-host-cpu-only qjoypad qubes-core-vchan-xen radicle-upstream rambox-pro renameutils rep riemann-c-client ripcord rkflashtool rrdtool sanoid sixpair slsnif smc smokeping solfege soulseekqt soundkonverter soundkonverter soundkonverter split2flac spring springlobby standardnotes station stdenv-linux super-productivity surf-display-unstable Sylk syslog-ng syslog-ng-incubator t1utils tang timedoctor timeular timidity tmpwatch tmux-cssh (20151015 → 2015-10-15) todoist-electron trezor-suite ttf-mkfontdir ttf2pt1 tusk-v0.23.0 udiskie unclutter unityhub uucp uudeview via vial vorbisgain wootility wv2 xdxf2slob-unstable xen xen xen xflux xflux-gui xtitle ytmdesktop yubico-piv-tool yubikey-personalization-gui zettlr zulip

and since that evaluates i think there is no issue in the syntax

i'm not sure if my method here is correct or the number ofborg found

**Comment by Stunkymonkey** [comment]
should i separate `ent`, `pv`, `pal`, `bfr`, `smc`, `wv2` and `unclutter` into a single PR for `staging`?
trying to get the rest merged into `master`?

**Comment by Stunkymonkey** [comment]
I think the number of mentions has to be done recursively, to get the number of how many packages are affected.
having the mentions is a good indicator, but can also mean nothing.

File: pkgs/tools/misc/dvtm/default.nix [hunk 1]
```diff
@@ -1,9 +1,10 @@
 {callPackage, fetchurl}:
 callPackage ./dvtm.nix rec {
-  name = "dvtm-0.15";
+  pname = "dvtm";
+  version = "0.15";
 
   src = fetchurl {
-    url = "http://www.brain-dump.org/projects/dvtm/${name}.tar.gz";
+    url = "http://www.brain-dump.org/projects/dvtm/dvtm-${version}.tar.gz";
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/dvtm/default.nix:7]:
```suggestion
    url = "https://www.brain-dump.org/projects/dvtm/dvtm-${version}.tar.gz";
```

File: pkgs/tools/misc/dvtm/unstable.nix [hunk 2]
```diff
@@ -1,6 +1,7 @@
 {callPackage, fetchFromGitHub, fetchpatch}:
 callPackage ./dvtm.nix {
-  name = "dvtm-unstable-2018-03-31";
+  pname = "dvtm-unstable";
+  version = "2018-03-31";
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/dvtm/unstable.nix:4]:
```suggestion
  pname = "dvtm";
  version = "unstable-2018-03-31";
```

File: pkgs/tools/misc/jdiskreport/default.nix [hunk 3]
```diff
@@ -10,11 +10,12 @@ let
...
 in
-stdenv.mkDerivation {
-  name = "jdiskreport-1.4.1";
+stdenv.mkDerivation rec {
+  pname = "jdiskreport";
+  version = "1.4.1";
 
   src = fetchurl {
-    url = "http://www.jgoodies.com/download/jdiskreport/jdiskreport-1_4_1.zip";
+    url = "http://www.jgoodies.com/download/jdiskreport/jdiskreport-${lib.replaceStrings ["."] ["_"] version}.zip";
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/jdiskreport/default.nix:18]:
```suggestion
    url = "https://www.jgoodies.com/download/jdiskreport/jdiskreport-${lib.replaceStrings ["."] ["_"] version}.zip";
```

File: pkgs/tools/misc/kisslicer/default.nix [hunk 4]
```diff
@@ -17,11 +17,12 @@ let
...
 
-stdenv.mkDerivation {
-  name = "kisslicer-1.6.3";
+stdenv.mkDerivation rec {
+  pname = "kisslicer";
+  version = "1.6.3";
 
   src = fetchzip {
-    url = "http://www.kisslicer.com/uploads/1/5/3/8/15381852/kisslicer_linux64_1.6.3_release.zip";
+    url = "http://www.kisslicer.com/uploads/1/5/3/8/15381852/kisslicer_linux64_${version}_release.zip";
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/kisslicer/default.nix:25]:
```suggestion
    url = "https://www.kisslicer.com/uploads/1/5/3/8/15381852/kisslicer_linux64_${version}_release.zip";
```

File: pkgs/tools/misc/logtop/default.nix [hunk 5]
```diff
@@ -1,7 +1,8 @@
 { lib, stdenv, fetchFromGitHub, ncurses, uthash, pkg-config }:
 
 stdenv.mkDerivation {
-  name = "logtop-0.7";
+  pname = "logtop";
+  version = "0.7";
 
   src = fetchFromGitHub {
     rev = "a0935ab2e7056feb3e8a90f5129990c9b335a587";
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/logtop/default.nix:8]:
```suggestion
    rev = "logtop-${version}";
```

File: pkgs/tools/misc/logtop/default.nix [hunk 6]
```diff
@@ -1,7 +1,8 @@
 { lib, stdenv, fetchFromGitHub, ncurses, uthash, pkg-config }:
 
 stdenv.mkDerivation {
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/logtop/default.nix:3]:
```suggestion
stdenv.mkDerivation rec {
```

File: pkgs/tools/misc/odroid-xu3-bootloader/default.nix [hunk 7]
```diff
@@ -1,7 +1,8 @@
 { stdenv, lib, fetchFromGitHub, coreutils, ubootOdroidXU3, runtimeShell }:
 
 stdenv.mkDerivation {
-  name = "odroid-xu3-bootloader-2015-12-04";
+  pname = "odroid-xu3-bootloader";
+  version = "2015-12-04";
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/odroid-xu3-bootloader/default.nix:5]:
```suggestion
  version = "unstable-2015-12-04";
```

File: pkgs/tools/misc/pg_top/default.nix [hunk 8]
```diff
@@ -1,10 +1,11 @@
 { lib, stdenv, fetchurl, ncurses, postgresql }:
 
 stdenv.mkDerivation rec {
-  name = "pg_top-3.7.0";
+  pname = "pg_top";
+  version = "3.7.0";
 
   src = fetchurl {
-    url = "http://pgfoundry.org/frs/download.php/1781/${name}.tar.gz";
+    url = "http://pgfoundry.org/frs/download.php/1781/pg_top-${version}.tar.gz";
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/pg_top/default.nix:8]:
```suggestion
    url = "https://pgfoundry.org/frs/download.php/1781/pg_top-${version}.tar.gz";
```

File: pkgs/tools/misc/sixpair/default.nix [hunk 9]
```diff
@@ -1,6 +1,7 @@
 { lib, stdenv, fetchurl, libusb-compat-0_1 }:
 stdenv.mkDerivation {
-  name = "sixpair-2007-04-18";
+  pname = "sixpair";
+  version = "2007-04-18";
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/sixpair/default.nix:4]:
```suggestion
  version = "unstable-2007-04-18";
```

File: pkgs/tools/misc/xflux/default.nix [hunk 10]
```diff
@@ -1,6 +1,7 @@
 {lib, stdenv, fetchurl, libXxf86vm, libXext, libX11, libXrandr, gcc}:
 stdenv.mkDerivation {
-  name = "xflux-2013-09-01";
+  pname = "xflux";
+  version = "2013-09-01";
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/xflux/default.nix:4]:
```suggestion
  version = "unstable-2013-09-01";
```

**Review by SuperSandro2000 — COMMENTED** [review]
also please target staging

File: pkgs/tools/misc/dvtm/unstable.nix (hunk 2 above)
Comment (Stunkymonkey) [reply pkgs/tools/misc/dvtm/unstable.nix:4]: i think this would be a breaking change. is it wanted?

Comment (SuperSandro2000) [reply pkgs/tools/misc/dvtm/unstable.nix:4]: right.

**Review by davidak — APPROVED** [review]
Changes look good

**Comment by davidak** [comment]
@Stunkymonkey thanks a lot for creating these PRs! :sparkles:
